### PR TITLE
update login and logout flow

### PR DESCRIPTION
**Pull request by EvaOyy**
Context: User feedback that they only realize that they need to login/signup on submitting their question. We should inform them before it.

Changes:
1) Bring users who are not logged in to the login page when they click on 'answer the question' button and 3 reaction buttons
2) Add Hi username when user logged in (need graphql help here - not done yet)

image

### CODE REVIEW

File: src/components/Navbar.js [hunk 1]
```diff
@@ -23,10 +23,11 @@ export default (props) => {
         style={{ backgroundColor: 'white' }}
       >
         <Nav style={{ alignItems: 'center' }}>
+          {authValue.isLoggedIn ? <NavLink>Hi, <u>User</u></NavLink> : <></>}
```
Comment (EvaOyy) [inline src/components/Navbar.js:26]: Need some guidance on how to create query for auth.User node

File: src/components/Navbar.js [hunk 2]
```diff
@@ -23,10 +23,11 @@ export default (props) => {
...
           <NavLink target='blank' href='https://forms.gle/tVzucdnhb6p5p2pRA'>Share Feedback</NavLink>
 
           {authValue.isLoggedIn
-            ? <NavLink onClick={handleLogOut}>Log Out</NavLink>
+            ? <NavLink href='/' onClick={handleLogOut}>Log Out</NavLink>
```
Comment (EvaOyy) [inline src/components/Navbar.js:30]: Bring user back to home page after logout

File: src/views/Login.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ const Login = ({ history }) => {
     required: ['username', 'password'],
     afterSubmit: res => {
       value.logIn(res.tokenAuth.token)
-      history.push('/')
+      history.goBack()
```
Comment (EvaOyy) [inline src/views/Login.js:35]: Bring user back to where they left off after login

**Review by jordantomax — CHANGES_REQUESTED** [review]
It's best if you based the PR off main, not off other open PRs. Otherwise, it's best to open only 1 PR as this one is dependent on the other. 

In any case, I'm not exactly clear on what this PR is trying to accomplish. When I click "Ask a Question", I'm not taken to Login as suggested by the description. Though I think that's not a great user experience anyway since it may discourage people from asking questions. I would prefer to have an integrated sign up flow with the ask question, like adding 2 form fields to ask question when user is logged out.

I think it's worth asking a designer about this as well cc @yeesiang-chng

**Review by EvaOyy — COMMENTED** [review]
Good catch, my PR missed out the 'Ask a question' button.
Currently only when user presses on the 3 reaction buttons or the Answer the question button will we bring them to the login page if they are not logged in yet.
This PR is merely a quick fix to bring user to login/Signup page before they can contribute to Verifact.

Agree that we can ask designer to suggest a better user flow.

**Pull request closed without merging**